src/stories/src/components/ConfirmedSearchInput.stories.tsx

```diff
@@ -0,0 +1,107 @@
+import type { Meta, StoryFn } from '@storybook/react';
+import React from 'react';
+import { ConfirmedSearchInput, TextArea } from 'src/components/input';
+
+/**
+ * This is a TextInput that fires its onChange callback when Enter, Escape, or
+ * the search button is triggered.
+ */
+const meta: Meta<typeof ConfirmedSearchInput> = {
+  title: 'src/Components/ConfirmedSearchInput',
+  component: ConfirmedSearchInput,
+  parameters: {
+    layout: 'centered',
+  },
+  tags: ['autodocs'],
+  argTypes: {
+    placeholder: {
+      control: 'text',
+      description: 'Placeholder text for when the input is empty',
+    },
+    defaultValue: {
+      control: 'text',
+      description: 'The initial value for the field',
+    },
+    value: {
+      control: false,
+      description: 'Value for the field',
+    },
+    disabled: {
+      control: 'boolean',
+      description: 'Should the input be disabled?',
+      table: {
+        defaultValue: { summary: 'false' },
+      },
+    },
+    onChange: {
+      type: 'function',
+      description: 'callback when the text is changed',
+    },
+    'aria-label': {
+      control: false,
+      description: 'Accessible label to use for the input if it is not linked to a label via the `id` attribute',
+    },
+    id: {
+      control: false,
+      description: 'Id to associate with the input. This should be linked to a label via `htmlFor`.',
+    },
+    autoFocus: {
+      control: 'boolean',
+      description: 'Should the input get initial focus on render?',
+      table: {
+        defaultValue: { summary: 'false' },
+      },
+    },
+    style: {
+      control: 'object',
+      description: 'Custom CSS styles for the input',
+      table: {
+        type: { summary: 'CSSProperties' },
+      },
+    },
+  },
+  args: {
+    defaultValue: 'default value',
+  },
+};
+
+export default meta;
+
+export const Example: StoryFn<typeof ConfirmedSearchInput> = (props) => {
+  const [_, updateArgs] = useArgs();
+  const onChange = (value: string) => {
+    updateArgs({ value });
+  };
+
+  return (
+    <div style={{ display: 'grid' }}>
+      {/* eslint-disable jsx-a11y/label-has-associated-control */}
+      <label style={{ marginBottom: '10px' }} htmlFor='search-input'>
+        Enter a search term:
+      </label>
+      <ConfirmedSearchInput
+        defaultValue={props.defaultValue}
+        placeholder={props.placeholder}
+        disabled={props.disabled}
+        onChange={onChange}
+        aria-label={props['aria-label']}
+        id='search-input'
+        autoFocus={props.autoFocus}
+        style={props.style}
+      />
+      {/* eslint-disable jsx-a11y/label-has-associated-control */}
+      <label
+        style={{
+          marginTop: '20px',
+          marginBottom: '10px',
+          fontStyle: 'italic',
+        }}
+        htmlFor='text-area'
+      >
+        onChange callback fired:
+      </label>
+      <TextArea value={props.value} id='text-area' disabled />
```

Comment:
only enter key seems to update this onChange display when I was playing with the control in Storybook...

Reply:
That's weird, it's not the experience I have when I use it in Storybook. Both Escape and pressing the Search field update the onChange entry (assuming that the text has changed from was previously evented). For example:

![image](https://github.com/DataBiosphere/terra-ui/assets/484484/2ee35241-f44a-4306-9110-f94a34e08bad)

<Press Escape>

![image](https://github.com/DataBiosphere/terra-ui/assets/484484/e3c74be7-180e-4431-a554-88090a0ff184)

I wonder if it's because you are on Windows?

Reply:
pressing tab or removing focus with mouse clicking something else did not work to trigger expected value change event

Reply:
@msilva-broad It isn't meant to… there are literally just these 3 things that fire an event: "This is a TextInput that fires its onChange callback when Enter, Escape, or the search button is triggered."

https://github.com/DataBiosphere/terra-ui/blob/706e93c89df2c604cb4fdd40aed546aa2fc0820d/src/components/input.js#L125

Now perhaps that isn't the behavior one would expect, but it's what we currently have.

Reply:
the curious thing is that the other controls work as expected.... which makes me think it's a missed bit of logic or event considerations inside this specific control... but not a blocker for this PR, just something to look into later.